Source/ARTRealtime.m

```diff
@@ -640,48 +643,48 @@ - (ARTEventListener *)transitionSideEffects:(ARTConnectionStateChange *)stateCha
             [self.auth cancelAuthorization:stateChange.reason];
             break;
         case ARTRealtimeDisconnected: {
-            if (!_startedReconnection) {
-                _startedReconnection = [NSDate date];
-                [_internalEventEmitter on:^(ARTConnectionStateChange *change) {
-                    if (change.current != ARTRealtimeDisconnected && change.current != ARTRealtimeConnecting) {
-                        self->_startedReconnection = nil;
-                    }
-                }];
-            }
-
-            if ([[NSDate date] timeIntervalSinceDate:_startedReconnection] >= _connectionStateTtl) {
-                artDispatchScheduled(0, _rest.queue, ^{
+            if ([self isSuspendMode]) {
+                [self.logger verbose:@"RT:%p realtime connection has staled", self];
+                [_connectionRetryFromDisconnectedListener stopTimer];
+                _connectionRetryFromDisconnectedListener = nil;
+                stateChangeEventListener = [self unlessStateChangesBefore:0 do:^{
```

Comment:
`0`? Why?

Reply:
It wasn't a nice solution. Changed on 928051b.